### ISSUE TITLE
midi pattern plugin GUI closes on any note insertion

### ISSUE THREAD

**Issue by FreeFull** [falkTX/Carla#950]
When I try to insert any note into the midi pattern, the window immediately disappears once I let go of the mouse button. When I reopen the GUI, the note is not there, so I can't actually use the midi pattern plugin at all.

Edit: Logs:

```
======= Starting engine =======
======= Engine started ========
Carla engine started, details:
  Driver name:  JACK
  Sample rate:  48000
  Process mode: Continuous Rack
qt.svg: <input>:7064: Could not resolve property: #radialGradient3709
Trying to start UI using "/usr/share/carla/resources/midipattern-ui"
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
ALL OK!
Trying to get window...
QPainter::begin: Painter already active
QPainter::begin: Painter already active
Trying to get window...
Match found using simple and UTF-8 name
QPainter::begin: Painter already active
Traceback (most recent call last):
  File "/usr/share/carla/resources/midipattern-ui", line 209, in sendMsg
    note_start = start * 60. / self.fTransportInfo["bpm"] * self.PPQ
ZeroDivisionError: float division by zero
CarlaPipeCommon::_writeMsgBuffer(..., 10) - failed with -1 (true), message was:
transport```

**Comment by falkTX** [comment]
This is happening because BPM is not being set, somehow ends up at 0.
Changing BPM manually will fix this.

**Comment by falkTX** [comment]
Fixed in d997f2976f7ed4035367b4601361521a4ed605e2

**Comment by falkTX** [comment]
I am closing this issue, since this particular bug is fixed.
But that plugin really needs some love so it works properly. I am in the middle of that, more to come.
Currently we have to set bpm to 60 when inputting notes, some of the calculations in there are wrong (likely was my fault).
Anyway, if you keep watching the git commits, eventually you will see the one that marks midi-pattern as a stable plugin. For now, that is not the case.
